rocketpy/Flight.py

```diff
@@ -2645,306 +2083,177 @@ def angleOfAttack(self, interpolation="spline", extrapolation="natural"):
 
         # Calculate angle of attack and convert to degrees
         angleOfAttack = np.rad2deg(np.arccos(dotProductNormalized))
-        angleOfAttack = np.column_stack([self.x[:, 0], angleOfAttack])
-
-        # Convert to Function
-        angleOfAttack = Function(
-            angleOfAttack,
-            "Time (s)",
-            "Angle Of Attack (°)",
-            interpolation,
-            extrapolation,
-        )
+        angleOfAttack = np.column_stack([self.time, angleOfAttack])
 
         return angleOfAttack
 
     # Stability and Control variables
+    def _frequency_analysis(self, func, samplingFrequency=1000):
+        """Frequency analysis of any function of time between out of rail time
+        and final simulation time.
+
+        Parameters
+        ----------
+        func : rocketpy.Function, callable
+            Function of time to be analyzed.
+
+        Returns
+        -------
+        FourierFrequencies : numpy.ndarray
+            Array of frequencies.
+        FourierAmplitude : numpy.ndarray
+            Array of amplitude values.
+        """
+        samplingTimeStep = 1.0 / samplingFrequency
+        samplingRange = np.arange(self.outOfRailTime, self.tFinal, samplingTimeStep)
+        sampledPoints = func(samplingRange)
+        sampledPoints -= np.mean(sampledPoints)
+        numberOfSamples = len(sampledPoints)
+        FourierAmplitude = np.abs(np.fft.fft(sampledPoints) / numberOfSamples)
+        FourierFrequencies = np.fft.fftfreq(numberOfSamples, samplingTimeStep)
+        return FourierFrequencies, FourierAmplitude
+
     # Angular velocities frequency response - Fourier Analysis
-    @cached_property
+    @funcify_method(
+        "Frequency (Hz)", "ω1 Angle Fourier Amplitude", "spline", "constant"
+    )
     def omega1FrequencyResponse(self):
-        Fs = 100.0
-        # sampling rate
-        Ts = 1.0 / Fs
-        # sampling interval
-        t = np.arange(1, self.tFinal, Ts)  # time vector
-        y = self.w1(t)
-        y -= np.mean(y)
-        n = len(y)  # length of the signal
-        k = np.arange(n)
-        T = n / Fs
-        frq = k / T  # two sides frequency range
```

Comment:
After this row, I can no longer understand the old (latest) implementation, therefore I cannot analyze what you (possibly) did wrong in your new proposal.